File: swap_skill_servver/src/main/java/com/example/swap_skill_servver/dto/SkillSwapRequestDto.java
package com.example.swap_skill_servver.dto;

import java.util.Set;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SkillSwapRequestDto {
    private UUID senderId;
    private UUID receiverId;
    private Set<String> senderTeachSkills;
    private Set<String> senderLearnSkills;
    private String message;
}

File: lib/features/request_skill/models/skill_swap_request_dto.dart
import 'dart:convert';

class SkillSwapRequestDto {
  final String senderId;
  final String receiverId;
  final List<String> senderTeachSkills;
  final List<String> senderLearnSkills;
  final String message;

  SkillSwapRequestDto({
    required this.senderId,
    required this.receiverId,
    required this.senderTeachSkills,
    required this.senderLearnSkills,
    required this.message,
  });

  Map<String, dynamic> toJson() {
    return {
      "senderId": senderId,
      "receiverId": receiverId,
      "senderTeachSkills": senderTeachSkills,
      "senderLearnSkills": senderLearnSkills,
      "message": message,
    };
  }

  String toRawJson() => json.encode(toJson());
}
